### PR TITLE
Conditionally set cursor graphic for unique items

### PR DESCRIPTION
**Pull request by kphoenix137**
Adds `ICURS_DEFAULT`, which when set for a unique item uses the base item's cursor graphic instead of setting its own.

Required to deal with Gnarled Root, a unique that is able to appear on 2 base items, but does not have an `INVCURS` item power in vanilla.

### CODE REVIEW

File: Source/itemdat.h [hunk 1]
```diff
@@ -282,6 +282,7 @@ enum item_cursor_graphic : uint8_t {
 	ICURS_BLITZEN                     = 219,
 	ICURS_DEMON_PLATE_ARMOR           = 225,
 	ICURS_BOVINE                      = 226,
+	ICURS_DEFAULT                     = -1,
```
Comment (DakkJaniels) [inline Source/itemdat.h:285]: You can't assign -1 to a `uint8_t`

Comment (kphoenix137) [reply Source/itemdat.h:285]: Yes you can, it underflows to 255

Comment (qndel) [reply Source/itemdat.h:285]: technically he can but it will be 255 xD

Comment (kphoenix137) [reply Source/itemdat.h:285]: Well I guess this is pissing off the checks, I'll have to change it :(

Comment (DakkJaniels) [reply Source/itemdat.h:285]: When I made the comment, it had already failed all the checks lol

Comment (kphoenix137) [reply Source/itemdat.h]: > When I made the comment, it had already failed ~all~ _most of_ the checks lol

I replied to you before I made it to the bottom of the page :D

Comment (DakkJaniels) [reply Source/itemdat.h]: maybe you could do something like `ICURS_DEFAULT  = 1 << (sizeof(item_cursor_graphic) - 1)`

Comment (kphoenix137) [reply Source/itemdat.h]: Donezo

File: Source/itemdat.h [hunk 2]
```diff
@@ -282,6 +282,8 @@ enum item_cursor_graphic : uint8_t {
 	ICURS_BLITZEN                     = 219,
 	ICURS_DEMON_PLATE_ARMOR           = 225,
 	ICURS_BOVINE                      = 226,
+
+	ICURS_DEFAULT  = 1 << (sizeof(item_cursor_graphic) - 1),
```
Comment (qndel) [inline Source/itemdat.h:286]: https://github.com/diasurgical/devilutionX/blob/039a6c91bc85e7106da94c4710fc3a305de3412f/Source/monster.h#L70
guess we should be fine with doing the same thing for consistency

**Comment by kphoenix137** [comment]
Wait, can't I just do what monstdat does for trn files and have empty fields instead which signal no iCurs change?

**Comment by AJenbo** [comment]
that sound somewhat nicer

**Comment by kphoenix137** [comment]
> that sound somewhat nicer, let me know when you figure out if you can

ready

**Review by AJenbo — APPROVED** [review]

File: Source/itemdat.h [hunk 3]
```diff
@@ -282,6 +282,8 @@ enum item_cursor_graphic : uint8_t {
 	ICURS_BLITZEN                     = 219,
 	ICURS_DEMON_PLATE_ARMOR           = 225,
 	ICURS_BOVINE                      = 226,
+
+	ICURS_DEFAULT = static_cast<uint8_t>(-1),
```
Comment (AJenbo) [inline Source/itemdat.h:286]:
```suggestion
	ICURS_DEFAULT = static_cast<uint8_t>(-1),
```

**Pull request merged**